src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(name) {
+    if ((name instanceof Herbivore) && (name.hidden === false)) {
+      name.health -= 50;
+    }
+
+    if (name.health === 0) {
+      Animal.alive = Animal.alive.filter(x => x.health !== 0);
```

Comment:
use descriptive names of variables instead of 'x'

Reply:
agree, pay attention to variable names